kernel/init.c

```diff
@@ -365,17 +374,26 @@ static void z_sys_init_run_level(enum init_level level)
 	const struct init_entry *entry;
 
 	for (entry = levels[level]; entry < levels[level+1]; entry++) {
-		const struct device *dev = entry->dev;
 		int result = 0;
 
+		if (unlikely(entry->_init_object == NULL)) {
```

Comment:
When does this occur? I'm drawing blanks for some reason. It would be good to add a comment.

Reply:
It's never supposed to occur actually, some prototyping left-over.

Reply:
Should it just be removed in that case? @gmarull - do you have any input here?

Reply:
I will remove it, hopefully an update today. There is a weird bug in some builds around the macros (which does not make much sense at first glance)